### PR TITLE
LUCENE-8915 : Improve Javadocs for RateLimiter and SimpleRateLimiter

### PR DESCRIPTION
**Pull request by atris**

### CODE REVIEW

File: lucene/core/src/java/org/apache/lucene/store/RateLimiter.java [hunk 1]
```diff
@@ -30,6 +30,8 @@
 
   /**
    * Sets an updated MB per second rate limit.
+   * A subclass is allowed to perform dynamic updates of the rate limit
+   * during use
```
Comment (sigram) [inline lucene/core/src/java/org/apache/lucene/store/RateLimiter.java]: Period. :)

File: lucene/core/src/java/org/apache/lucene/store/RateLimiter.java [hunk 2]
```diff
@@ -47,7 +49,11 @@
    * */
   public abstract long pause(long bytes) throws IOException;
   
-  /** How many bytes caller should add up itself before invoking {@link #pause}. */
+  /** How many bytes caller should add up itself before invoking {@link #pause}.
+   *  NOTE: The value returned by this method may change over time and is not guaranteed
+   *  to be constant throughout the lifetime of the RateLimiter. Users are advised to
+   *  refresh their local values with calls to this method to ensure consistency
```
Comment (sigram) [inline lucene/core/src/java/org/apache/lucene/store/RateLimiter.java]: Period. :)

Comment (sigram) [reply lucene/core/src/java/org/apache/lucene/store/RateLimiter.java]: Overall, LGTM, +1.

**Comment by atris** [comment]
@sigram Thanks, updated the same

**Comment by atris** [comment]
@sigram does this look commit right now, please? Happy to iterate further

**Pull request merged**